Panic when running prune on repository with empty snapshot file

restic 0.13.1 compiled with go1.18.1 on linux/amd64

Today I accidentally disconnected the USB drive containing my restic repository after backup, before the kernel was done writing. After removing a partially written index file[^1], restic now panics when pruning. I'm creating this issue as I guess it should exit with an error message, and that the panic might point to a bug somewhere. *I have another copy of my backups so I'm not looking for a solution for the corruption.*

The panic:
```
[root@laptop BACKUP]# restic -r ./repository prune
enter password for repository:
repository bbde6d6c opened successfully, password is correct
loading indexes...
loading all snapshots...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x55cc97214394]

goroutine 84 [running]:
main.getUsedBlobs.func1({0xb3, 0x5b, 0x83, 0x0, 0x66, 0xa, 0xcc, 0xbf, 0xa4, 0x28, ...}, ...)
        github.com/restic/restic/cmd/restic/cmd_prune.go:558 +0x54
github.com/restic/restic/internal/restic.ForAllSnapshots.func2()
        github.com/restic/restic/internal/restic/snapshot.go:113 +0x185
golang.org/x/sync/errgroup.(*Group).Go.func1()
        golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57 +0x67
created by golang.org/x/sync/errgroup.(*Group).Go
        golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x8d
```

<details><summary>Extra context about how I got here</summary>
<div>
After unplugging my disk, prune failed with an error:

```
[root@laptop BACKUP]# restic -r ./repository prune
enter password for repository:
repository bbde6d6c opened successfully, password is correct
loading indexes...
Fatal: load <index/7f8b7bf231>: invalid data returned
```

So I removed the index file it was referencing:

```
[root@laptop BACKUP]# rm repository/index/7f8b7bf231f0949810d347db5a15b5dde5392219bbbed329b40e15536572aa78
```

At this point I expected `restic prune` to either:

* Show an error similar to the last one.
* Prune the leftover unused files.

But it now panics as shown at the top of this issue.
</div>
</details>

[^1]: Hoping that restic would just ignore the extra data and forget about my last snapshot.

--- a/cmd/restic/cmd_prune.go
+++ b/cmd/restic/cmd_prune.go
@@ -596,10 +596,11 @@ func getUsedBlobs(gopts GlobalOptions, repo restic.Repository, ignoreSnapshots r
 	Verbosef("loading all snapshots...\n")
 	err = restic.ForAllSnapshots(gopts.ctx, repo.Backend(), repo, ignoreSnapshots,
 		func(id restic.ID, sn *restic.Snapshot, err error) error {
-			debug.Log("add snapshot %v (tree %v, error %v)", id, *sn.Tree, err)
 			if err != nil {
+				debug.Log("failed to load snapshot %v (error %v)", id, err)
 				return err
 			}
+			debug.Log("add snapshot %v (tree %v)", id, *sn.Tree)
 			snapshotTrees = append(snapshotTrees, *sn.Tree)
 			return nil
 		})

--- a/cmd/restic/cmd_prune.go
+++ b/cmd/restic/cmd_prune.go
@@ -605,7 +605,7 @@ func getUsedBlobs(gopts GlobalOptions, repo restic.Repository, ignoreSnapshots r
 			return nil
 		})
 	if err != nil {
-		return nil, err
+		return nil, errors.Fatalf("failed loading snapshot: %v", err)
 	}
 
 	Verbosef("finding data that is still in use for %d snapshots\n", len(snapshotTrees))

